commit 50d30e203b35
Author: Jim Chen <nchen@mozilla.com>
Date:   Mon May 1 14:46:00 2017 -0400

    Bug 1357874 - Add more AArch64 support to JS code; r=luke
    
    * Fix a parentheses warning when compiling testGCAllocator.cpp.
    
    * Enable trace logging in TraceLogging.cpp.
---
 js/src/jsapi-tests/testGCAllocator.cpp |  2 +-
 js/src/jsmath.cpp                      |  2 ++
 js/src/jsnativestack.cpp               | 10 +++++-----
 js/src/vm/TraceLogging.cpp             |  2 +-
 js/src/wasm/WasmSignalHandlers.cpp     |  2 ++
 5 files changed, 11 insertions(+), 7 deletions(-)

diff --git js/src/jsapi-tests/testGCAllocator.cpp js/src/jsapi-tests/testGCAllocator.cpp
index 229e56422fdf..ec5407e8cca1 100644
--- js/src/jsapi-tests/testGCAllocator.cpp
+++ js/src/jsapi-tests/testGCAllocator.cpp
@@ -315,7 +315,7 @@ mapMemoryAt(void* desired, size_t length)
 
 #if defined(__ia64__) || defined(__aarch64__) || \
     (defined(__sparc__) && defined(__arch64__) && (defined(__NetBSD__) || defined(__linux__)))
-    MOZ_RELEASE_ASSERT(0xffff800000000000ULL & (uintptr_t(desired) + length - 1) == 0);
+    MOZ_RELEASE_ASSERT((0xffff800000000000ULL & (uintptr_t(desired) + length - 1)) == 0);
 #endif
     void* region = mmap(desired, length, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANON, -1, 0);
     if (region == MAP_FAILED)
diff --git js/src/vm/TraceLogging.cpp js/src/vm/TraceLogging.cpp
index c7bc6a84b2e9..82fdf7c77e8a 100644
--- js/src/vm/TraceLogging.cpp
+++ js/src/vm/TraceLogging.cpp
@@ -61,7 +61,7 @@ rdtsc(void)
     return result;
 
 }
-#elif defined(__arm__)
+#elif defined(__arm__) || defined(__aarch64__)
 
 #include <sys/time.h>
 
